tests/integration/index/array_test.go

```diff
@@ -0,0 +1,1103 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package index
```

Comment:
praise: These tests are very readable and it made understanding the feature and reviewing the PR much easier,  thanks Islam :)